lib/galaxy/authnz/psa_authnz.py

```diff
@@ -497,3 +500,53 @@ def disconnect(
     # option B
     # user_authnz.extra_data = None
     sa_session.commit()
+
+
+def decode_access_token(social: UserAuthnzToken, backend: OpenIdConnectAuth, **kwargs):
+    """
+    Auth pipeline step to decode the OIDC access token.
+
+    Returns the access token, making it available as a new argument
+    "access_token" that can be used in future pipeline steps
+
+    Depends on "access_token" being present in social.extra_data,
+    which should be handled by social_core.pipeline.social_auth.load_extra_data, so
+    this step should be placed after load_extra_data in the pipeline.
+    """
+    access_token_encoded = social.extra_data.get("access_token")
+    if access_token_encoded is None:
+        return {"access_token": None}
+    if not _is_decodable_jwt(access_token_encoded):
+        log.info("Access token is not in header.payload.signature format and can't be decoded (may be an opaque token)")
+        return {"access_token": None}
+    access_token_data = _decode_access_token_helper(token_str=access_token_encoded, backend=backend)
+    return {"access_token": access_token_data}
+
+
+def _is_decodable_jwt(token_str: str) -> bool:
```

Comment:
Is this guaranteed to work for opaque tokens? I was wondering, should we instead just try decoding the jwt and ignoring it if undecodable instead?